### ISSUE TITLE
AttributeError: module 'typing' has no attribute 'TypedDict'

### ISSUE THREAD

**Issue by schlegelp** [clbarnes/extreqs#1]
Hi Chris. `navis` tests on GH actions have started running into issues during install (see [here](https://github.com/navis-org/navis/actions/runs/4302143563/jobs/7500243478). I suppose this has something to do with release of `1.1.0`?

```bash 
2023-03-01T09:43:06.7690600Z   × Getting requirements to build editable did not run successfully.
2023-03-01T09:43:06.7691382Z   │ exit code: 1
2023-03-01T09:43:06.7787311Z   ╰─> [21 lines of output]
2023-03-01T09:43:06.7787593Z       Traceback (most recent call last):
2023-03-01T09:43:06.7795329Z         File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
2023-03-01T09:43:06.7795773Z           main()
2023-03-01T09:43:06.7796418Z         File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
2023-03-01T09:43:06.7796922Z           json_out['return_val'] = hook(**hook_input['kwargs'])
2023-03-01T09:43:06.7797636Z         File "/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 132, in get_requires_for_build_editable
2023-03-01T09:43:06.7798051Z           return hook(config_settings)
2023-03-01T09:43:06.7798656Z         File "/tmp/pip-build-env-8zlupj9p/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 447, in get_requires_for_build_editable
2023-03-01T09:43:06.7799098Z           return self.get_requires_for_build_wheel(config_settings)
2023-03-01T09:43:06.7799722Z         File "/tmp/pip-build-env-8zlupj9p/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
2023-03-01T09:43:06.7800279Z           return self._get_build_requires(config_settings, requirements=['wheel'])
2023-03-01T09:43:06.7800878Z         File "/tmp/pip-build-env-8zlupj9p/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
2023-03-01T09:43:06.7801525Z           self.run_setup()
2023-03-01T09:43:06.7802246Z         File "/tmp/pip-build-env-8zlupj9p/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 335, in run_setup
2023-03-01T09:43:06.7802618Z           exec(code, locals())
2023-03-01T09:43:06.7802893Z         File "<string>", line 5, in <module>
2023-03-01T09:43:06.7803404Z         File "/tmp/pip-build-env-8zlupj9p/overlay/lib/python3.7/site-packages/extreqs/__init__.py", line 4, in <module>
2023-03-01T09:43:06.7803834Z           from .main import parse_requirement_files, parse_requirement_files_dict
2023-03-01T09:43:06.7804402Z         File "/tmp/pip-build-env-8zlupj9p/overlay/lib/python3.7/site-packages/extreqs/main.py", line 127, in <module>
2023-03-01T09:43:06.7804774Z           class ParsedFiles(tp.TypedDict):
2023-03-01T09:43:06.7805148Z       AttributeError: module 'typing' has no attribute 'TypedDict'
2023-03-01T09:43:06.7805422Z       [end of output]
2023-03-01T09:43:06.7805615Z   
```

**Comment by clbarnes** [comment]
Ah yes, there's a new feature which makes it marginally easier to unpack the output straight into setup args, which uses typeddict. I'll fix the python version requirements in the project metadata, which should make the dependency resolver pick an older version of extreqs - python 3.7 is EOL in 3 months anyway (and dropped by numpy releases in 2021) so shouldn't be an issue for long.

**Comment by schlegelp** [comment]
Thanks! I should probably stop testing against and supporting 3.7 too

**Comment by clbarnes** [comment]
I've released 1.0.1 with the fixed python version requirements and yanked 1.0.0 so that the dependency resolver doesn't try to fall back to that. Should work now! Thanks for raising it.

**Comment by clbarnes** [comment]
1.1.0 reinstates python 3.7 support by putting the TypedDict definition behind a version check.
